src/views/ReferralProgram.vue

```diff
@@ -0,0 +1,384 @@
+          {{ t('referralProgram.action.copyLink') }}
+          <s-icon name="copy-16" size="16" />
+        </s-button>
+      </s-card>
+      <p
+        v-else
+        class="referral-program-hint referral-program-hint--connected"
+        v-html="t('referralProgram.startInviting')"
+      />
+    </template>
+    <template v-else>
+      <p class="referral-program-hint" v-html="t('referralProgram.connectAccount')" />
+      <s-button
+        v-if="!(loading || isSoraAccountConnected)"
+        class="connect-button s-typography-button--large"
+        type="primary"
+        @click="handleConnect"
+      >
+        {{ t('referralProgram.action.connectWallet') }}
+      </s-button>
+    </template>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins } from 'vue-property-decorator';
+import { Action, Getter } from 'vuex-class';
+import { components, mixins } from '@soramitsu/soraneo-wallet-web';
+import { KnownSymbols } from '@sora-substrate/util/build/assets/consts';
+import type { AccountAsset } from '@sora-substrate/util/build/assets/types';
+
+import router, { lazyComponent, lazyView } from '@/router';
+import { PageNames, Components, LogoSize } from '@/consts';
+import { copyToClipboard } from '@/utils';
+
+import WalletConnectMixin from '@/components/mixins/WalletConnectMixin';
+
+const namespace = 'referrals';
+
+@Component({
+  components: {
+    FormattedAmount: components.FormattedAmount,
+    InfoLine: components.InfoLine,
+    TokenLogo: lazyComponent(Components.TokenLogo),
+    ReferralBonding: lazyView(PageNames.ReferralBonding),
+  },
+})
+export default class ReferralProgram extends Mixins(
+  mixins.LoadingMixin,
+  mixins.TransactionMixin,
+  mixins.FormattedAmountMixin,
+  mixins.ReferralRewardsMixin,
+  WalletConnectMixin
+) {
+  readonly LogoSize = LogoSize;
+
+  @Getter('tokenXOR', { namespace: 'assets' }) tokenXOR!: AccountAsset;
+  @Getter('invitedUsers', { namespace }) invitedUsers!: Array<string>;
+
+  @Action('getInvitedUsers', { namespace }) getInvitedUsers!: (referralId: string) => Promise<void>;
+  @Action('setBound', { namespace }) setBound!: (isBond: boolean) => Promise<void>;
+
+  get rewards(): string {
+    return this.referralRewards?.rewards.toLocaleString() || '0';
+  }
+
+  get invitedUserRewards(): any {
+    return this.referralRewards?.invitedUserRewards;
+  }
+
+  get bondedXOR(): string {
+    return this.tokenXOR?.balance?.bonded || '';
+  }
+
+  get xorSymbol(): string {
+    return KnownSymbols.XOR;
+  }
+
+  get invitedUsersNumber(): number {
+    return this.invitedUsers.length;
+  }
+
+  get invitedUsersClasses(): Array<string> {
+    const baseClass = 'invited-users-container';
+    const cssClasses: Array<string> = [baseClass];
+    if (!this.invitedUsersNumber) {
+      cssClasses.push('is-active');
+      cssClasses.push(`${baseClass}--empty`);
+    }
+    return cssClasses;
+  }
+
+  get referralLink(): any {
+    const polkaswapLink = window.location.origin;
```

Comment:
Could you please use `detectBaseUrl` from '@/api' as this util return correct `base` using ipfs link?